examples/ssh/linux_iplatency.js

```diff
@@ -0,0 +1,107 @@
+ * Communication protocol is SSH
+ * 
+ * Tested on Linux Distributions:
+ *      - Ubuntu 20.04 LTS
+ * Shell Version:
+ *      - Bash 5.1.16
+ * 
+ * Creates a Custom Driver table with a list of IP addresses, their average latency, and their packet loss.
+ * 
+ **/
+
+var pktno = "2"; // Number of packets to send during the ping command.
+var ipAddresses = ["8.8.8.8", "1.1.1.1", "192.168.0.1", "192.168.0.1", "172.17.0.1"]; // List of IP addresses to ping and retrieve status for.
+
+// Set up the SSH command options
+var sshCommandOptions = {
+    "timeout": 30000
+};
+
+var tableColumns = D.createTable(
+    "IP Latency",
+    [
+        { label: "IP Address" },
+        { label: "Latency", unit: "ms" },
+        { label: "Packet Loss", unit: "%" }
+    ]
+);
+
+// A function to check for authentication errors during execution
+function checkSshError(err) {
+    if (err.message) console.error(err.message);
+    if (err.code == 5) D.failure(D.errorType.AUTHENTICATION_ERROR);
+    console.error(err);
+    D.failure(D.errorType.GENERIC_ERROR);
+}
+
+function executeCommand(command){
+    var d = D.q.defer();
+    sshCommandOptions.command = command;
+    D.device.sendSSHCommand(sshCommandOptions, function (output, error) {
+        if(error) checkSshError(error);
+        d.resolve(output);
+    });
+    return d.promise;
+}
+
+/**
+ * Excuting a simple command to test access to device:
+ * 'ls' list the files and directories within a specified directory
+ * @remote_procedure
+ * @label Validate Association
+ * @documentation This procedure is used to validate if the driver can be applied on a device during association as well as validate any credentials provided
+ */
+function validate() {
+    var commandValidate = "ls";
+    console.info("Verifying credentials ... ", commandValidate); 
+    executeCommand(commandValidate)
+        .then(function(){
+            D.success();
+        })
+        .catch(function(error) {
+            checkSshError(error);
+        });
+}
+
+/**
+ * @remote_procedure
+ * @label Get IP Latency
+ * @documentation This procedure retrieves the latency of each IP address by sending ping commands.
+ * It populates the Custom Driver table with the IP address, latency, and packet loss.
+ */
+function get_status() {
+    var commandes  = ipAddresses.map(function (ipAddress) {
+        console.info("Pinging " + ipAddress + " ... ");
+        var command = "ping -c " + pktno + " " + ipAddress;
+        return executeCommand(command)
+            .then(function (output) {
+                parseOutput(output, ipAddress);
+            })
+            .catch(function (error) {
+                checkSshError(error);
+            });
+    });
+      
+    D.q.all(commandes)
+        .then(function () {
+            D.success(tableColumns);
+        })
+        .catch(function (error) {
+            console.error(error);
+            D.failure(D.errorType.GENERIC_ERROR);
+        });
+}
+
+function parseOutput(output, ipAddress){
```

Comment:
You are not handling error cases for a single address (for example 100% packet loss)

{"code":1,"output":"PING 8.8.8.33 (8.8.8.33) 56(84) bytes of data.\n\n--- 8.8.8.33 ping statistics ---\n2 packets transmitted, 0 received, 100% packet loss, time 1003ms"}